Source/santa-driver/SantaCache.h

```diff
@@ -186,6 +185,35 @@ template<typename KeyT, typename ValueT> class SantaCache {
     return count_;
   }
 
+  /**
+    Fill in the per_bucket_counts array with the number of entries in each bucket.
+
+    The per_buckets_count array will contain the per-bucket counts, up to the number
+    in array_size. The start_bucket parameter will determine which bucket to start off
+    with and upon return will contain either 0 if no buckets are remaining or the next
+    bucket to begin with when called again.
+  */
+  void bucket_counts(uint16_t *per_bucket_counts, uint16_t *array_size, uint64_t *start_bucket) {
```

Comment:
Any reason to not just return a vector<uint16_t> directly?

Reply:
No vector in the kernel, sadly.